src/NzbDrone.Core/Download/Clients/DownloadStation/DownloadStation.cs

```diff
@@ -0,0 +1,323 @@
+        protected readonly IDownloadStationProxy _proxy;
+        protected readonly ISharedFolderResolverService _sharedFolderResolver;
+
+        public DownloadStation(IDownloadStationProxy proxy,
+                               ITorrentFileInfoReader torrentFileInfoReader,
+                               IHttpClient httpClient,
+                               IConfigService configService,
+                               IDiskProvider diskProvider,
+                               IRemotePathMappingService remotePathMappingService,
+                               Logger logger,
+                               ICacheManager cacheManager,
+                               ISharedFolderResolverService sharedFolderResolver)
+            : base(torrentFileInfoReader, httpClient, configService, diskProvider, remotePathMappingService, logger)
+        {
+            _proxy = proxy;
+            _sharedFolderResolver = sharedFolderResolver;
+            _sharedFolderResolver.Settings = Settings;
+        }
+
+        public override string Name => "Download Station";
+
+        protected virtual OsPath GetOutputPath(OsPath outputPath, DownloadStationTorrent torrent)
+        {
+            return outputPath + torrent.Title;
+        }
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            IEnumerable<DownloadStationTorrent> torrents;
+
+            try
+            {
+                torrents = _proxy.GetTorrents(Settings);
+            }
+            catch (DownloadClientException ex)
+            {
+                _logger.Error(ex);
+                return Enumerable.Empty<DownloadClientItem>();
+            }
+
+            var items = new List<DownloadClientItem>();
+
+            foreach (var torrent in torrents)
+            {
+                var outputPath = new OsPath($"/{torrent.Additional.Detail["destination"]}");
+
+                if (Settings.TvDirectory.IsNotNullOrWhiteSpace())
+                {                                                 
+                    if (!outputPath.Contains(new OsPath(Settings.TvDirectory)))
+                    {
+                        continue;
+                    }
+                }
+                else if (Settings.TvCategory.IsNotNullOrWhiteSpace())
+                {
+                    var directories = outputPath.FullPath.Split('\\', '/');
+                    if (!directories.Contains(Settings.TvCategory))
+                    {
+                        continue;
+                    }
+                }
+
+                var physicalPath = _sharedFolderResolver.ResolvePhysicalPath(outputPath.FullPath);
+
+                outputPath = _remotePathMappingService.RemapRemoteToLocal(Settings.Host, outputPath);
+                
+                var item = new DownloadClientItem()
+                {
+                    Category = Settings.TvCategory,
+                    DownloadClient = Definition.Name,
+                    DownloadId = torrent.Id,
```

Comment:
The id is somesthing like "task_xx" where xx is an incremental integer number. Let me see if i can get some other info from the diskstation that i can use, a serial number or something like that...